BFB restartability

dazlich opened the pull request:
ocn_comp_nuopc.F90 had its import of coupler fields fixed and this now establishes bit-for-bit restartability of the restart files for mpas-ocean. Some modifications to defaults for diagnostic output were made so that restarted runs produce bit-for-bit reproducible output as well for some of the diagnostics. Monthly mean time average output is not yet reproducible as the first time step after initialization fails to accumulate

dazlich committed:
Bit-for-bit restartability established for mpas-ocean

ocn_comp_nuopc.F90 had its import of coupler fields fixed
and this now establishes bit-for-bit restartability of the
restart files for mpas-ocean. Some modifications to defaults for
diagnostic output were made so that restarted runs produce
bit-for-bit reproducible output as well for some of the
diagnostics. Monthly mean time average output is not yet reproducible
as the first time step after initialization fails to accumulate

dazlich committed:
Modify the imported swnet check

Fixing bit-for-bit restartability required adding an additional
import from the coupler at initialization. It turns out that
not all fluxes were defined at this point and this tripped
up the check for positive SW in this added call. These changes
allow this check to be skipped for this additional initial import.

gdicker1 commented:
@dazlich, I think this may need some more looking. I could only get Intel to run when I tried these changes. I used `create_test` for exact restart tests using the CHAOS2000dev compset at 120km using the Intel, NVHPC, and GNU compilers. Both NVHPC and GNU appear to abort during init, before the first timestep.

NVHPC aborts with repeated print outs in cesm.log.\* like "`ERROR: gcell,shortWaveHeatFlux =             1    79.22896428346886`" and eventually "`ERROR: (set_surface_forcing) ERROR: SHF_QSW < qsw_eps in set_surface_forcing`". There were no log.{ocean,seaice}*.err files.

GNU aborts without much print out to cesm.log. It creates MPAS-O error log files, named similarly to "log.ocean.0000.d****.err" (literal '*' characters). These error log files contain the same "`CRITICAL ERROR: ERROR in ocn_export`" line which is also at the end of the ocn.log.\* file.

My attempts are in sub-dirs of "/glade/derecho/scratch/gdicker/2025Jan21_EWPR_BfBRestarts_99"

dazlich commented:
I thought I had that fixed - I’ll tackle that again - thanks.

On Jan 21, 2025, at 6:50 PM, G. Dylan Dickerson ***@***.***> wrote:


** Caution: EXTERNAL Sender **


@dazlich<https://github.com/dazlich>, I think this may need some more looking. I could only get Intel to run when I tried these changes. I used create_test for exact restart tests using the CHAOS2000dev compset at 120km using the Intel, NVHPC, and GNU compilers. Both NVHPC and GNU appear to abort during init, before the first timestep.

NVHPC aborts with repeated print outs in cesm.log.* like "ERROR: gcell,shortWaveHeatFlux = 1 79.22896428346886" and eventually "ERROR: (set_surface_forcing) ERROR: SHF_QSW < qsw_eps in set_surface_forcing". There were no log.{ocean,seaice}*.err files.

GNU aborts without much print out to cesm.log. It creates MPAS-O error log files, named similarly to "log.ocean.0000.d****.err" (literal '*' characters). These error log files contain the same "CRITICAL ERROR: ERROR in ocn_export" line which is also at the end of the ocn.log.* file.

My attempts are in sub-dirs of "/glade/derecho/scratch/gdicker/2025Jan21_EWPR_BfBRestarts_99"

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/mpas-ocean/pull/19#issuecomment-2606107287>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWBC4O5LWPATPSRMH3L2L32PXAVCNFSM6AAAAABVDLKZFOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMBWGEYDOMRYG4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

dazlich committed:
Fixed two issues on the road to BFB restartability

Two issues were uncovered and fixed on the road to
bit-for-bit restartability:
1) The nvhpc compiler failed the positive SW test in
subroutine ocn_import. This test was rewritten to avoid
the use of the fortran 'ANY' function and now works.
2) The gnu compiler failed because an errorcode had not
been initialized. The proper initialization was added and
now works.
